src/vs/workbench/services/notebook/common/notebookDocumentService.ts

```diff
@@ -79,17 +79,18 @@ export function extractCellOutputDetails(uri: URI): { notebook: URI; openIn: str
 	const outputId = params.get('outputId') ?? undefined;
 	const parsedCell = parse(uri.with({ scheme: Schemas.vscodeNotebookCell, query: null }));
 	const outputIndex = params.get('outputIndex') ? parseInt(params.get('outputIndex') || '', 10) : undefined;
-
-	if (parsedCell?.notebook === undefined || parsedCell?.handle === undefined) {
-		throw new Error('Invalid cell URI');
-	}
+	const notebookUri = parsedCell ? parsedCell.notebook : uri.with({
```

Comment:
If the `parsedCell` is invalid, then doesn't this mean the cell no longer exists?
Also shouldn't we then just return `undefined `here, instead of returning something invalid by making up the details, as that could lead to other issues.
I.e. returning `undefined` instead of throwing an error might be better.

Reply:
like @amunger said, for the old style of open output in a text editor, we don't have the cell fragment in the URI just the output ID. We then want to return the notebook but do not mind that the cell fragment doesn't exist in this instance.